spyder/plugins/ipythonconsole.py

```diff
@@ -1050,15 +1046,194 @@ def additional_options(self):
+        instance_font_style = give_font_style(ins_fs)
+
+        import pygments
+        syntax_path = pygments.__file__.rpartition('/')[0] + '/styles/'
+        syntax_name = 'spyder'
+        syntax_filename = syntax_path + syntax_name + '.py'
+
+        syntax_style_imports = ['Name', 'Keyword', 'Comment', 'String',
+                                'Number', 'Punctuation']
+        imports_string = ", ".join(syntax_style_imports)
+
+        syntax_style_dic = {}
+        syntax_style_dic['Name'] = [font_font_style, font_font_weight,
+                                    font_color]
+        syntax_style_dic['Name.Class'] = [definition_font_style,
+                                          definition_font_weight,
+                                          definition_color]
+        syntax_style_dic['Name.Function'] = [definition_font_style,
+                                             definition_font_weight,
+                                             definition_color]
+        syntax_style_dic['Name.Builtin'] = [builtin_font_style,
+                                            builtin_font_weight, builtin_color]
+        syntax_style_dic['Name.Builtin.Pseudo'] = [instance_font_style,
+                                                   instance_font_weight,
+                                                   instance_color]
+        syntax_style_dic['Keyword'] = [keyword_font_style, keyword_font_weight,
+                                       keyword_color]
+        syntax_style_dic['Keyword.Type'] = [builtin_font_style,
+                                            builtin_font_weight, builtin_color]
+        syntax_style_dic['Comment'] = [comment_font_style, comment_font_weight,
+                                       comment_color]
+        syntax_style_dic['String'] = [string_font_style, string_font_weight,
+                                      string_color]
+        syntax_style_dic['Number'] = [number_font_style, number_font_weight,
+                                      number_color]
+        syntax_style_dic['Punctuation'] = [font_font_style, font_font_weight,
+                                           font_color]
+
+        syntax_file_string = "# -*- coding: utf-8 -*-\n"
+        syntax_file_string += "from pygments.style import Style\n"
+        syntax_file_string += "from pygments.token import {}\n\n".format(imports_string)
+        syntax_file_string += "class SpyderStyle(Style):\n"
+        syntax_file_string += "\tstyles = {\n"
+        for key in syntax_style_dic:
+            syntax_file_string += "\t\t{}:'".format(key)
+            values = syntax_style_dic[key]
+            for val in values:
+                if val != '':
+                    syntax_file_string += "{} ".format(val)
+            syntax_file_string += "',\n"
+        syntax_file_string += "\t}"
+
+        with open(syntax_filename, "w") as syntax_file:
+            syntax_file.write(syntax_file_string)
+            syntax_file.close()

```

Comment:
You don't need to close a file if you're using a context manager (i.e. a `with` command). That's the beauty of it.